### PR TITLE
Add pixel geolocation utility

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

**Comment by JonathanMeans** [comment]
Hang on, missed something tricky in the offset part.

**Comment by JonathanMeans** [comment]
Good to go

File: six/modules/c++/samples/geolocate_pixel.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        const bool shadowsDown(options->get<bool>("shadowsDown"));
+        if (!six::sicd::AreaPlaneUtility::hasAreaPlane(*complexData))
+        {
+            six::sicd::AreaPlaneUtility::setAreaPlane(*complexData);
+        }
+
+        six::sicd::AreaPlane& plane = *complexData->radarCollection->area->plane;
+        if (shadowsDown)
+        {
+            plane.rotateCCW();
```
Comment (asylvest) [inline six/modules/c++/samples/geolocate_pixel.cpp]: You can't just always rotate CCW if you want shadows down... you need to check the orientation to see if you should rotate.  Probably worth adding a `rotateToShadowsDown()` method to the plane that does this check for you.  Note that if the orientation is `ARBITRARY`, you won't want to rotate it at all

**Comment by JonathanMeans** [comment]
Made the changes.

File: six/modules/c++/samples/sicd_output_plane_pixel_to_lat_lon.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ * Provide a pixel coordinate and a SICD. This program will output the
+ * latitude, longitude, and latitude of the pixel in the output plane.
```
Comment (asylvest) [inline six/modules/c++/samples/sicd_output_plane_pixel_to_lat_lon.cpp]: second 'latitude' should be 'altitude'

File: six/modules/c++/samples/sicd_output_plane_pixel_to_lat_lon.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        if (!sys::OS().exists(sicdPathname))
+        {
+            std::cerr << sicdPathname << " does not exist.\n";
+            return 1;
+        }
+
+        std::vector<std::string> schemaPaths;
+        getSchemaPaths(*options, "--schemaPath", "schema", schemaPaths);
+
+        const double col = options->get<double>("col");
```
Comment (asylvest) [inline six/modules/c++/samples/sicd_output_plane_pixel_to_lat_lon.cpp]: Leftover `col`

Comment (JonathanMeans) [reply six/modules/c++/samples/sicd_output_plane_pixel_to_lat_lon.cpp]: Could this line have been from an outdated version? I don't see it...

Comment (asylvest) [reply six/modules/c++/samples/sicd_output_plane_pixel_to_lat_lon.cpp]: OK yeah I think the GitHub UI confused me or something - I agree that it's gone.

**Comment by asylvest** [comment]
Looks good!

**Pull request merged**